configs/config.go

```diff
@@ -15,11 +16,27 @@ type AppConfig struct {
 
 var appConfig *AppConfig
 
-func GetAppConfig() *AppConfig {
-	err := godotenv.Load(".env")
+const projectDirName = "testGoApi"
+
+func loadEnv() {
+	projectName := regexp.MustCompile(`^(.*` + projectDirName + `)`)
+	currentWorkDirectory, err := os.Getwd()
+
+	if err != nil {
+		log.Fatal("Failed to get current working directory", err)
+	}
+
+	rootPath := projectName.Find([]byte(currentWorkDirectory))
+
+	err = godotenv.Load(string(rootPath) + `/.env`)
+
 	if err != nil {
 		log.Fatalf("Error loading .env file")
 	}
+}
+
+func GetAppConfig() *AppConfig {
+	loadEnv()
```

Comment:
this should be called inside the `if appConfig == nil {` so that file reading will happen only once